core/domain/html_validation_service.py

```diff
@@ -703,3 +703,31 @@ def is_parsable_as_xml(xml_string):
         return True
     except xml.etree.ElementTree.ParseError:
         return False
+
+
+def convert_svg_diagram_tags_to_image_tags(html_string):
+    """Renames all the oppia-noninteractive-svgdiagram on the
+       server to oppia-noninteractive-image and changes
+       corresponding attributes
+
+    Args:
+        html_string: str. The HTML string to check.
+
+    Returns:
+        str. The updated html string.
+    """
+    soup = bs4.BeautifulSoup(
+        html_string.encode(encoding='utf-8'), 'html.parser')
+
+    for image in soup.findAll(name='oppia-noninteractive-svgdiagram'):
+        # The 'filepath-with-value' and 'alt-with-value'
+        # attribute values should be enclosed in
+        # double quotes(&amp;quot;).
+        attr_value = escape_html(image['svg_filename-with-value'])
+        alt_value = escape_html(image['alt-with-value'])
+        del image['svg_filename-with-value']
```

Comment:
It might be more correct to remove all attributes, instead, so that only the necessary attributes for noninteractive-image are added.

Reply:
done.

Reply:
Ah, I also just realized that image isn't actually a dictionary so it can't quite be cleared the way I was hoping.